src/servers/src/query_handler.rs

```diff
@@ -45,7 +46,13 @@ pub type ScriptHandlerRef = Arc<dyn ScriptHandler + Send + Sync>;
 
 #[async_trait]
 pub trait SqlQueryHandler {
-    async fn do_query(&self, query: &str, query_ctx: QueryContextRef) -> Result<Output>;
+    async fn do_query(&self, query: &str, query_ctx: QueryContextRef) -> Vec<Result<Output>>;
+
+    async fn do_statement_query(
```

Comment:
WDYT this alternative: Remove `do_query` from `SqlQueryHandler`, let each server decide how to parse the raw string query. It comes with 2 benifits in that: 1. server could choose dialect, instead of the only one currently used "GenericDialect"; 2. simplify trait implementors, since these 2 method are mostly logically the same, doesn't need to repeat yourself. If it seems good to you, we may file an issue and resolve it in some following PR.

Reply:
Good idea. Sounds good to me.

Reply:
We need to check if `Statement` generated from different dialect are exact same. Otherwise we might not be able to share a common handler and it's going to be complex to maintain two or more different stacks.